### PR TITLE
feat(core) permissions configurer

### PR DESCRIPTION
**Pull request by danielpeach**
@anotherchrisberry @ttomsu please review.
permissions_configurer
attributes

### CODE REVIEW

File: app/scripts/modules/core/src/application/config/applicationAttributes.directive.html [hunk 1]
```diff
@@ -26,9 +26,10 @@
   <dt ng-if="vm.application.attributes.legacyUdf">User Data Format</dt>
   <dd ng-if="vm.application.attributes.legacyUdf">This application requires legacy user data format.</dd>
   <dt ng-if="vm.application.attributes.requiredGroupMembership.length">Group Membership</dt>
```
Comment (anotherchrisberry) [inline app/scripts/modules/core/src/application/config/applicationAttributes.directive.html]: Did you mean mean to leave this line in and remove the one below?

File: app/scripts/modules/core/src/application/modal/PermissionsConfigurer.spec.tsx [hunk 2]
```diff
@@ -0,0 +1,56 @@
+import {mock} from 'angular';
+import * as React from 'react';
+import {mount} from 'enzyme';
+import {ReactInjector, REACT_MODULE} from 'core/reactShims';
+
+import {
+  IPermissions, IPermissionsConfigurerProps,
+  PermissionsConfigurer
+} from 'core/application/modal/PermissionsConfigurer';
+import {AUTHENTICATION_SERVICE} from 'core/authentication';
```
Comment (anotherchrisberry) [inline app/scripts/modules/core/src/application/modal/PermissionsConfigurer.spec.tsx]: space around `{` and `}`

(I know it's hard to keep up with the ever-changing code format standards)

File: app/scripts/modules/core/src/application/config/applicationAttributes.directive.html (hunk 1 above)
Comment (danielpeach) [inline app/scripts/modules/core/src/application/config/applicationAttributes.directive.html]: I meant to remove both - thanks.

**Comment by gardleopard** [comment]
Looks great!

File: app/scripts/modules/core/src/application/modal/PermissionsConfigurer.tsx [hunk 3]
```diff
@@ -0,0 +1,219 @@
+import * as React from 'react';
+import * as Select from 'react-select';
+import {Button} from 'react-bootstrap';
+import {ReactInjector} from 'core/reactShims';
```
Comment (jrsquared) [inline app/scripts/modules/core/src/application/modal/PermissionsConfigurer.tsx]: `import {} from 'core/...'` goes after all the external imports with a space between them all.

File: app/scripts/modules/core/src/application/modal/PermissionsConfigurer.tsx [hunk 4]
```diff
@@ -0,0 +1,219 @@
...
+    if (this.props.permissions && this.props.permissions.WRITE) {
+      WRITE = this.props.permissions.WRITE.slice();
+    }
+
+    this.props.requiredGroupMembership.forEach(group => {
+      READ.push(group);
+      WRITE.push(group);
+    });
+
+    uniq(READ);
```
Comment (jrsquared) [inline app/scripts/modules/core/src/application/modal/PermissionsConfigurer.tsx]: doesn't `uniq` create a new duplicate-free array and preserve the old array?

Comment (danielpeach) [inline app/scripts/modules/core/src/application/modal/PermissionsConfigurer.tsx]: Whoops. Thanks.

**Pull request merged**